lib/core/src/Cardano/Wallet/DB/Layer.hs

```diff
@@ -800,37 +759,47 @@ newDBLayerWith _cacheBehavior _tr ti SqliteContext{runQuery} = do
 
         , updatePendingTxForExpiry = \wid tip -> ExceptT $ do
             selectWallet wid >>= \case
-                Nothing -> pure $ Left $ ErrNoSuchWallet wid
-                Just _ -> Right <$> updatePendingTxForExpiryQuery wid tip
-
-        , removePendingOrExpiredTx = \wid tid -> ExceptT $ do
-            let errNoSuchWallet =
-                    Left $ ErrRemoveTxNoSuchWallet $ ErrNoSuchWallet wid
-            let errNoMorePending =
-                    Left $ ErrRemoveTxAlreadyInLedger tid
-            let errNoSuchTransaction =
-                    Left $ ErrRemoveTxNoSuchTransaction $
-                    ErrNoSuchTransaction wid tid
+                Nothing -> pure $ Left  $ ErrNoSuchWallet wid
+                Just _ -> modifyDBMaybe transactionsDBVar $ \_ ->
+                    let
+                        delta = Just
+                            $ ChangeTxMetaWalletsHistory wid
+                            $ AgeTxMetaHistory tip
+                    in  (delta, Right ())
+        , removePendingOrExpiredTx = \wid txId -> ExceptT $ do
             selectWallet wid >>= \case
-                Nothing -> pure errNoSuchWallet
-                Just _  -> selectTxMeta wid tid >>= \case
-                    Nothing -> pure errNoSuchTransaction
-                    Just _ -> do
-                        count <- deletePendingOrExpiredTx wid tid
-                        pure $ if count == 0
-                            then errNoMorePending
-                            else Right ()
+                Nothing -> pure $ Left
+                    $ ErrRemoveTxNoSuchWallet $ ErrNoSuchWallet wid
+                Just _ -> do
+                    ex <- modifyDBMaybe transactionsDBVar
+                        $ \(TxHistoryF txsOld, _ws) ->
+                            if TxId txId `Map.member` txsOld
+                                then
+                                    let
+                                        delta = Just
+                                            $ ChangeTxMetaWalletsHistory wid
+                                            $ PruneTxMetaHistory $ TxId txId
+                                    in  (delta, Right ())
+                                else (Nothing, Left $ ErrRemoveTxNoSuchTransaction
+                                    $ ErrNoSuchTransaction wid txId)
+                    pure $ case ex of
+                        Left ert -> case ert of
+                            ErrRemoveTxNoSuchWallet _ ->
```

Comment:
```suggestion
                            -- Fixup error message in the case where
                            -- the wallet exists but contains no transactions.
                            ErrRemoveTxNoSuchWallet _ ->
```

Reply:
The case is not needed, I'm a fool! The DBVar will not fail on its own with random messages (NoSucTx etc), in fact we were checking this in explicitly before at this layer!